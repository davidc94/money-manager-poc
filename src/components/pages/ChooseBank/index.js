import React, { PureComponent } from 'react';
import {
  View,
  TouchableOpacity,
  Image,
  FlatList,
  StyleSheet,
} from 'react-native';
import { connect } from 'react-redux';
import StyledButton from '../../atoms/StyledButton';
import StyledText from '../../atoms/StyledText';
import { FETCH_BANK_DATA } from "../../../store/constants";
import bankImages from '../../../assets/images/banklogos/index';

const styles = StyleSheet.create({
  bankItem: {
    flexDirection: 'row',
    padding: 20,
    marginTop: 10,
    marginHorizontal: 10,
    borderWidth: 1,
  },
});

class ChooseBank extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      selectedBank: false,
    }
  }

  componentDidMount(){
    this.props.dispatch({ type: FETCH_BANK_DATA })
  }

  selectBank = (selectedBank) => this.setState({ selectedBank });

  renderListItem = ({ item }) => {
    const bdColor = item.bankName === this.state.selectedBank ? '#000' : '#ccc';



    return (
      <TouchableOpacity
        onPress={() => this.selectBank(item.bankName)}
        style={[styles.bankItem, { borderColor: bdColor }]}
        data-test="bank-list-item"
      >
        <View style={{ width: 50, height: 50, margin: 5, flex: 1 }}>
          <Image style={{ width: '100%', height: '100%', resizeMode: 'contain' }} source={bankImages[item.bankName]} />
        </View>
        <StyledText style={{ flex: 4 }}>
          {item.bankName}
        </StyledText>
      </TouchableOpacity>
    )
  }
  render() {
    return (
      <View>
        <StyledText style={{ marginVertical: 30 }}>
          Select your bank that your day to day spending account is with.
        </StyledText>
        <FlatList
          style={{ marginBottom: 50 }}
          data={this.props.bankList}
          extraData={this.state}
          renderItem={this.renderListItem}
          keyExtractor={(item) => item.bankName}
        />
        <View style={{ paddingHorizontal: 80 }}>
          <StyledButton
            disabled={!this.state.selectedBank}
            onPress={() => this.props.navigation.navigate('Consent')}
            data-test="nextButton"
          >
            Connect to bank
          </StyledButton>
        </View>
      </View >
    );
  }
}

const mapStateToProps = (state) => ({
  bankList: state.bankReducers.bankList
})

export default connect (mapStateToProps)(ChooseBank);
