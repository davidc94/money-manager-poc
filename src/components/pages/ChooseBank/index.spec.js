import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ChooseBank from './index';

configure({ adapter: new Adapter() });

describe('choose bank', () => {
  const render = shallow(<ChooseBank />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});