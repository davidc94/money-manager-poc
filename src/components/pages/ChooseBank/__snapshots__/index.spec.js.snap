// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`choose bank renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChooseBank />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StyledText
          style={
            Object {
              "marginVertical": 30,
            }
          }
        >
          Select your bank that your day to day spending account is with.
        </StyledText>,
        <FlatList
          data={
            Array [
              Object {
                "bankName": "Barclays",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/barclayslogo.jpg",
                },
              },
              Object {
                "bankName": "LloydsBank",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/lloydslogo.jpg",
                },
              },
              Object {
                "bankName": "HSBC",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/hsbclogo.jpg",
                },
              },
            ]
          }
          disableVirtualization={false}
          extraData={
            Object {
              "data": Array [
                Object {
                  "bankName": "Barclays",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/barclayslogo.jpg",
                  },
                },
                Object {
                  "bankName": "LloydsBank",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/lloydslogo.jpg",
                  },
                },
                Object {
                  "bankName": "HSBC",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/hsbclogo.jpg",
                  },
                },
              ],
              "selectedBank": false,
            }
          }
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          style={
            Object {
              "marginBottom": 50,
            }
          }
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
        <Component
          style={
            Object {
              "paddingHorizontal": 80,
            }
          }
        >
          <StyledButton
            disabled={true}
            onPress={[Function]}
          >
            Connect to bank
          </StyledButton>
        </Component>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Select your bank that your day to day spending account is with.",
          "style": Object {
            "marginVertical": 30,
          },
        },
        "ref": null,
        "rendered": "Select your bank that your day to day spending account is with.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "bankName": "Barclays",
              "logoImage": Object {
                "testUri": "../../../src/assets/images/banklogos/barclayslogo.jpg",
              },
            },
            Object {
              "bankName": "LloydsBank",
              "logoImage": Object {
                "testUri": "../../../src/assets/images/banklogos/lloydslogo.jpg",
              },
            },
            Object {
              "bankName": "HSBC",
              "logoImage": Object {
                "testUri": "../../../src/assets/images/banklogos/hsbclogo.jpg",
              },
            },
          ],
          "disableVirtualization": false,
          "extraData": Object {
            "data": Array [
              Object {
                "bankName": "Barclays",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/barclayslogo.jpg",
                },
              },
              Object {
                "bankName": "LloydsBank",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/lloydslogo.jpg",
                },
              },
              Object {
                "bankName": "HSBC",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/hsbclogo.jpg",
                },
              },
            ],
            "selectedBank": false,
          },
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "style": Object {
            "marginBottom": 50,
          },
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <StyledButton
            disabled={true}
            onPress={[Function]}
          >
            Connect to bank
          </StyledButton>,
          "style": Object {
            "paddingHorizontal": 80,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Connect to bank",
            "disabled": true,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "Connect to bank",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StyledText
            style={
              Object {
                "marginVertical": 30,
              }
            }
          >
            Select your bank that your day to day spending account is with.
          </StyledText>,
          <FlatList
            data={
              Array [
                Object {
                  "bankName": "Barclays",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/barclayslogo.jpg",
                  },
                },
                Object {
                  "bankName": "LloydsBank",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/lloydslogo.jpg",
                  },
                },
                Object {
                  "bankName": "HSBC",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/hsbclogo.jpg",
                  },
                },
              ]
            }
            disableVirtualization={false}
            extraData={
              Object {
                "data": Array [
                  Object {
                    "bankName": "Barclays",
                    "logoImage": Object {
                      "testUri": "../../../src/assets/images/banklogos/barclayslogo.jpg",
                    },
                  },
                  Object {
                    "bankName": "LloydsBank",
                    "logoImage": Object {
                      "testUri": "../../../src/assets/images/banklogos/lloydslogo.jpg",
                    },
                  },
                  Object {
                    "bankName": "HSBC",
                    "logoImage": Object {
                      "testUri": "../../../src/assets/images/banklogos/hsbclogo.jpg",
                    },
                  },
                ],
                "selectedBank": false,
              }
            }
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            style={
              Object {
                "marginBottom": 50,
              }
            }
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
          <Component
            style={
              Object {
                "paddingHorizontal": 80,
              }
            }
          >
            <StyledButton
              disabled={true}
              onPress={[Function]}
            >
              Connect to bank
            </StyledButton>
          </Component>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Select your bank that your day to day spending account is with.",
            "style": Object {
              "marginVertical": 30,
            },
          },
          "ref": null,
          "rendered": "Select your bank that your day to day spending account is with.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "bankName": "Barclays",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/barclayslogo.jpg",
                },
              },
              Object {
                "bankName": "LloydsBank",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/lloydslogo.jpg",
                },
              },
              Object {
                "bankName": "HSBC",
                "logoImage": Object {
                  "testUri": "../../../src/assets/images/banklogos/hsbclogo.jpg",
                },
              },
            ],
            "disableVirtualization": false,
            "extraData": Object {
              "data": Array [
                Object {
                  "bankName": "Barclays",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/barclayslogo.jpg",
                  },
                },
                Object {
                  "bankName": "LloydsBank",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/lloydslogo.jpg",
                  },
                },
                Object {
                  "bankName": "HSBC",
                  "logoImage": Object {
                    "testUri": "../../../src/assets/images/banklogos/hsbclogo.jpg",
                  },
                },
              ],
              "selectedBank": false,
            },
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "style": Object {
              "marginBottom": 50,
            },
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <StyledButton
              disabled={true}
              onPress={[Function]}
            >
              Connect to bank
            </StyledButton>,
            "style": Object {
              "paddingHorizontal": 80,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Connect to bank",
              "disabled": true,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "Connect to bank",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
